### ISSUE TITLE
iocage list -PR returns 'HTTPResponse' object has no attribute 'geturl'

### ISSUE THREAD

**Issue by ccammack** [iocage/iocage#994]
I installed iocage using the binary package (pkg install py36-iocage) and get this error when trying to list the plugins.

```bash
$ su
Password:

# iocage activate zroot
ZFS pool 'zroot' successfully activated.

# iocage list --plugins --remote
'HTTPResponse' object has no attribute 'geturl'

# iocage list -PR
'HTTPResponse' object has no attribute 'geturl'

# iocage list --plugins --remote --http
'HTTPResponse' object has no attribute 'geturl'

# iocage list -PR --http
'HTTPResponse' object has no attribute 'geturl'

# iocage -v
Version 1.1 RELEASE 2019/01

# uname -a
FreeBSD freebsd 12.0-RELEASE-p7 FreeBSD 12.0-RELEASE-p7 GENERIC  amd64

# ping github.com
PING github.com (192.30.255.112): 56 data bytes
64 bytes from 192.30.255.112: icmp_seq=0 ttl=54 time=27.605 ms
64 bytes from 192.30.255.112: icmp_seq=1 ttl=54 time=33.778 ms
64 bytes from 192.30.255.112: icmp_seq=2 ttl=54 time=27.865 ms
64 bytes from 192.30.255.112: icmp_seq=3 ttl=54 time=27.227 ms
^C
--- github.com ping statistics ---
4 packets transmitted, 4 packets received, 0.0% packet loss
round-trip min/avg/max/stddev = 27.227/29.119/33.778/2.700 ms
```

**Comment by sonicaj** [comment]
Hello @ccammack if i am not wrong, can you please verify if your aim is to list available plugins ?
If yes, then please just use
```
iocage fetch --plugins
```
This will show plugins which are available. `R` flag represents something else so `PR` gives unexpected results as expected.
Closing this issue, please feel free to comment here if any more help is required.
Thank you!

**Comment by sonicaj** [comment]
So @ccammack i just tried this and it actually works for me. This is interesting, can you please confirm if this happens for you in latest master version ? ( i checked in master )

**Comment by ccammack** [comment]
Thanks for the quick response @sonicaj.

Yes, I was trying to get a list of available plugins, following the docs for 1.2 RC at https://iocage.readthedocs.io/en/latest/plugins.html

> To see a list of all currently available plugins, open a command line and type iocage list -PR or iocage list --plugins --remote. The full iocage plugin list is also available on GitHub.

> Check which plugins are installed on the system with iocage list --plugins or iocage list -P

I just tried fetch without -R:

```bash
# iocage fetch --plugins
An IP address is needed to fetch a plugin!
Please specify ip(4|6)_addr="[INTERFACE|]IPADDRESS"!

# iocage fetch --plugins ip4_addr="em0"
'HTTPResponse' object has no attribute 'geturl'
```

I also tried list without -R:

```bash
# iocage list --plugins
+-----+------+------+-------+------+---------+-----+-----+----------+--------+
| JID | NAME | BOOT | STATE | TYPE | RELEASE | IP4 | IP6 | TEMPLATE | PORTAL |
+=====+======+======+=======+======+=========+=====+=====+==========+========+
+-----+------+------+-------+------+---------+-----+-----+----------+--------+
```

I will install the source tree and try the master this evening.

**Comment by ccammack** [comment]
Okay, installing the master allows iocage list to work. Thanks for the help.

```
# iocage -v
Version 1.2 RC
# iocage list -PR
Branch 12.0-RELEASE does not exist at https://github.com/freenas/iocage-ix-plugins.git!
Using "master" branch for plugin, this may not work with your RELEASE
+-------------------+-------------------+-------------------+------------------+
|       NAME        |    DESCRIPTION    |        PKG        |       ICON       |
+===================+===================+===================+==================+
| BackupPC          | BackupPC is a     | backuppc          | https://www.true |
|                   | high-performance, |                   | os.org/iocage-ic |
|                   | enterprise-grade  |                   | ons/backuppc.png |
|                   | system for        |                   |                  |
|                   | backing up Linux, |                   |                  |
|                   | WinXX and MacOSX  |                   |                  |
|                   | PCs and laptops   |                   |                  |
|                   | to a server's     |                   |                  |
|                   | disk.             |                   |                  |
+-------------------+-------------------+-------------------+------------------+
| Bacula-server     | programs to       | bacula-server     | https://www.true |
|                   | manage backup,    |                   | os.org/iocage-   |
|                   | recovery, and     |                   | icons/bacula-    |
|                   | verification of   |                   | server.png       |
|                   | computer data     |                   |                  |
|                   | across a network  |                   |                  |
|                   | of computers of   |                   |                  |
|                   | different kinds   |                   |                  |
+-------------------+-------------------+-------------------+------------------+
...
```
